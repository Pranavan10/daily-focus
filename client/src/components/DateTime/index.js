import React, { useEffect, useState } from "react";
import styles from "./style.module.scss";

function DateTime() {
    const [dateObject, setDateObject] = useState(new Date());

    useEffect(() => {
        const interval = setTimeout(() => setDateObject(new Date(), 1000 * 60));
        return () => clearTimeout(interval);
    }, [dateObject]);

    const time = dateObject.toLocaleString("en-US", {
        hour: "numeric",
        minute: "numeric",
        hour12: true,
    });

    const dayArray = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
    const monthArray = [
        "January",
        "February",
        "March",
        "April",
        "May",
        "June",
        "July",
        "August",
        "September",
        "October",
        "November",
        "December",
    ];
    let day = dayArray[dateObject.getDay()];
    let date = dateObject.getDate();
    let month = monthArray[dateObject.getMonth()];

    return <div className={styles.text}>{` ${time} | ${day} ${date} ${month}`}</div>;
}

export default DateTime;
