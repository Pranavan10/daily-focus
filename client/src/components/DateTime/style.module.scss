.text {
    color: #7d7d7d;
    font-family: "Ropa Sans", sans-serif;
    font-weight: 400;
    font-size: 1.6rem;
}
