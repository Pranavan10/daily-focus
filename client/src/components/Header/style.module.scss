.header {
    display: flex;
    height: 100px;
    justify-content: space-around;
    align-items: center;
    font-size: 1.5rem;
    background-color: white;
    -webkit-box-shadow: 0px 7px 20px 0px rgba(0, 0, 0, 0.1);
    -moz-box-shadow: 0px 7px 20px 0px rgba(0, 0, 0, 0.1);
    box-shadow: 0px 7px 20px 0px rgba(0, 0, 0, 0.1);

    .focusLogo {
        flex: 1 1 0;
        text-align: center;
    }

    .dateTime {
        flex: 1 1 0;
    }
    .searchBar {
        flex: 1.5 1 0;
    }

    .addWidget {
        flex: 0.75 1 0;
        text-align: center;
    }

    .userIcon {
        flex: 0.4 2 0;
    }
}
