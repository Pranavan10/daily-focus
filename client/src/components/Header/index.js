import React from "react";
import Button from "@material-ui/core/Button";
import { IconButton, makeStyles } from "@material-ui/core";
import PersonIcon from "@material-ui/icons/Person";
import styles from "./style.module.scss";
import FocusLogo from "../../images/focus-logo.svg";
import SearchBar from "../SearchBar";
import DateTime from "../DateTime";

const useStyles = makeStyles({
    root: {
        background: "#30A0F5",
        borderRadius: 50,
        padding: "4px 25px",
    },
    label: {
        textTransform: "capitalize",
        fontSize: "1.25rem",
    },
    icon: {
        "background": "#30A0F5",
        "padding": 10,
        "&:hover": {
            backgroundColor: "#303f9f",
        },
    },
    iconLabel: {
        fill: "white",
        fontSize: "2rem",
    },
});

export default function Header() {
    const classes = useStyles();

    return (
        <header className={styles.header}>
            <div className={styles.focusLogo}>
                <img style={{ width: 175 }} src={FocusLogo} alt="FOCUS" />
            </div>

            <div className={styles.dateTime}>
                <DateTime />
            </div>

            <div className={styles.searchBar}>
                <SearchBar />
            </div>

            <div className={styles.addWidget}>
                <Button
                    variant="contained"
                    color="primary"
                    classes={{ root: classes.root, label: classes.label }}
                >
                    Add Widget
                </Button>
            </div>

            <div className={styles.userIcon}>
                <IconButton aria-label="user menu" classes={{ root: classes.icon }}>
                    <PersonIcon classes={{ root: classes.iconLabel }} />
                </IconButton>
            </div>
        </header>
    );
}
