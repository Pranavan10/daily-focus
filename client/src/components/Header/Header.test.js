import React from "react";
import { shallow } from "enzyme";
import ShallowRenderer from "react-test-renderer/shallow";

import Header from ".";
import DateTime from "../DateTime";
import SearchBar from "../SearchBar";

let component;

beforeEach(() => {
    component = shallow(<Header />);
});

test("it renders correctly", () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshotComponent = shallowRenderer.render(<Header />);
    expect(snapshotComponent).toMatchSnapshot();
});

test("it contains a DateTime component", () => {
    expect(component.find(DateTime)).toHaveLength(1);
});

test("it contains a SearchBar component", () => {
    expect(component.find(SearchBar)).toHaveLength(1);
});
