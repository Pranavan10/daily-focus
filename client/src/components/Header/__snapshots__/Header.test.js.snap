// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="focusLogo"
  >
    <img
      alt="FOCUS"
      src="focus-logo.svg"
      style={
        Object {
          "width": 175,
        }
      }
    />
  </div>
  <div
    className="dateTime"
  >
    <DateTime />
  </div>
  <div
    className="searchBar"
  >
    <SearchBar />
  </div>
  <div
    className="addWidget"
  >
    <WithStyles(ForwardRef(Button))
      classes={
        Object {
          "label": "makeStyles-label-2",
          "root": "makeStyles-root-1",
        }
      }
      color="primary"
      variant="contained"
    >
      Add Widget
    </WithStyles(ForwardRef(Button))>
  </div>
  <div
    className="userIcon"
  >
    <WithStyles(ForwardRef(IconButton))
      aria-label="user menu"
      classes={
        Object {
          "root": "makeStyles-icon-3",
        }
      }
    >
      <Memo
        classes={
          Object {
            "root": "makeStyles-iconLabel-4",
          }
        }
      />
    </WithStyles(ForwardRef(IconButton))>
  </div>
</header>
`;
